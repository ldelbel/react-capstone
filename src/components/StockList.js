import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import StockCard from './StockCard';
import Filter from './Filter';
import updateFilter from '../actions/index';
import mapStockstoProps from '../utils/index';

const StyledContainer = styled.div`
border-radius: 20px;
padding: 0;
margin-top: 35px;
box-shadow: 2px 2px 8px 2px rgba(33,37,41,0.62);
`;

const StockList = props => {
  const { list } = props;
  console.log(list);
  const handleUpdateFilter = target => {
    props.updateFilter(target);
  };
  if (list) {
    return (
      <StyledContainer className="container">
        <Filter handleUpdateFilter={handleUpdateFilter} />
        <div
          className="d-flex flex-row justify-content-start flex-wrap"
          style={{
            background: '#17A2B8', minHeight: '90px', marginTop: '0', borderBottomLeftRadius: '20px', borderBottomRightRadius: '20px', paddingRight: '0', paddingTop: '0', paddingBottom: '15px', paddingLeft: '45px',
          }}
        >
          {list.map(stock => (
            <StockCard key={stock.symbol} profile={stock} />
          ))}
        </div>
      </StyledContainer>
    );
  }
  return (
    <div>Loading</div>
  );
};

StockList.propTypes = {
  list: PropTypes.arrayOf(
    PropTypes.shape({
      symbol: PropTypes.string.isRequired,
    }),
  ).isRequired,
  updateFilter: PropTypes.func.isRequired,
};

export default connect(mapStockstoProps, { updateFilter })(StockList);
