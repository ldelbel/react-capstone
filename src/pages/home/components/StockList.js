import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import StockCard from './StockCard';
import Filter from './Filter';
import LoadingDiv from '../../common/Loading';
import updateFilter from '../../../actions/index';
import mapStockstoProps from '../../../utils/index';

const StyledContainer = styled.div`
border-radius: 20px;
padding: 0;
margin-top: 35px;
box-shadow: 2px 2px 8px 2px rgba(33,37,41,0.62);
`;

const StockList = props => {
  const { list } = props;
  const handleUpdateFilter = target => {
    props.updateFilter(target);
  };
  return (
    <StyledContainer className="container">
      <Filter handleUpdateFilter={handleUpdateFilter} />
      <div
        className="d-flex flex-row justify-content-start flex-wrap"
        style={{
          background: '#17A2B8', minHeight: '90px', marginTop: '0', borderBottomLeftRadius: '20px', borderBottomRightRadius: '20px', paddingRight: '0', paddingTop: '10px', paddingBottom: '15px', paddingLeft: '45px',
        }}
      >
        {list
          ? (
            list.map(stock => (
              <StockCard key={stock.symbol} profile={stock} />
            ))
          )
          : <LoadingDiv />}
      </div>
    </StyledContainer>
  );
};

StockList.propTypes = {
  list: PropTypes.arrayOf(
    PropTypes.shape({
      symbol: PropTypes.string.isRequired,
    }),
  ).isRequired,
  updateFilter: PropTypes.func.isRequired,
};

export default connect(mapStockstoProps, { updateFilter })(StockList);
