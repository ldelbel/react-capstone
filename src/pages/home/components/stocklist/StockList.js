import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import StockCard from '../stockcard/StockCard';
import Filter from '../filter/Filter';
import LoadingDiv from '../../../../common/components/Loading';
import updateFilter from '../../../../actions/index';
import mapStockstoProps from '../../../../utils/index';
import styles from './StockList.module.scss';

const StyledContainer = styled.div`
border-radius: 20px;
padding: 0;
margin-top: 35px;
box-shadow: 2px 2px 8px 2px rgba(33,37,41,0.62);
`;

const StockList = props => {
  const { list } = props;
  const handleUpdateFilter = target => {
    props.updateFilter(target);
  };
  return (
    <StyledContainer className="container">
      <Filter handleUpdateFilter={handleUpdateFilter} />
      <div
        className={`d-flex flex-row justify-content-start flex-wrap ${styles.stocklist}`}
      >
        {list
          ? (
            list.map(stock => (
              <StockCard key={stock.symbol} profile={stock} />
            ))
          )
          : <LoadingDiv />}
      </div>
    </StyledContainer>
  );
};

StockList.propTypes = {
  list: PropTypes.arrayOf(
    PropTypes.shape({
      symbol: PropTypes.string.isRequired,
    }),
  ).isRequired,
  updateFilter: PropTypes.func.isRequired,
};

export default connect(mapStockstoProps, { updateFilter })(StockList);
