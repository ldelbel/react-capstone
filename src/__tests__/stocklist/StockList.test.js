import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import StockList from '../../components/StockList';

it('renders correctly', () => {
  const navbar = renderer.create(
    <BrowserRouter>
      <StockList />
    </BrowserRouter>,
  ).toJSON();
  expect(navbar).toMatchSnapshot();
});