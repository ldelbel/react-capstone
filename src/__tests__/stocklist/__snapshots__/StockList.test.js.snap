// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when store is provided 1`] = `
<div
  className="sc-iBPRYJ fmVwbk container"
>
  <div
    style={
      Object {
        "background": "#343a40",
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": 120,
        "padding": 15,
        "paddingLeft": 30,
        "paddingRight": 30,
      }
    }
  >
    <div>
      <span
        style={
          Object {
            "color": "var(--white)",
            "fontFamily": "\\"Alfa Slab One\\", cursive",
            "fontSize": 30,
          }
        }
      >
        Filters 
      </span>
    </div>
    <div
      className="d-flex flex-row justify-content-center align-items-center"
      style={
        Object {
          "height": 50,
        }
      }
    >
      <div
        style={
          Object {
            "marginRight": 20,
          }
        }
      >
        <span
          style={
            Object {
              "color": "var(--light)",
              "fontFamily": "Alata, sans-serif",
              "fontSize": 20,
              "marginRight": 10,
            }
          }
        >
          Price Range 
        </span>
        <select
          defaultValue="null"
          name="price"
          onChange={[Function]}
          style={
            Object {
              "borderRadius": 8,
            }
          }
        >
          <option
            value="null"
          >
            Select
          </option>
          <option
            value={100}
          >
            &lt; 100
          </option>
          <option
            value={200}
          >
            &lt; 200
          </option>
          <option
            value={300}
          >
            &lt; 300
          </option>
          <option
            value={500}
          >
            &lt; 500
          </option>
          <option
            value={1000}
          >
            &lt; 1000
          </option>
          <option
            value={2000}
          >
            &lt; 2000
          </option>
        </select>
      </div>
      <div
        style={
          Object {
            "marginRight": 20,
          }
        }
      >
        <span
          style={
            Object {
              "color": "var(--light)",
              "fontFamily": "Alata, sans-serif",
              "fontSize": 20,
              "marginRight": 10,
            }
          }
        >
          Sector 
        </span>
        <select
          defaultValue="null"
          name="sector"
          onChange={[Function]}
          style={
            Object {
              "borderRadius": 8,
            }
          }
        >
          <option
            value="null"
          >
            Select
          </option>
          <option
            value="Aerospace & Defense"
          >
            Aerospace & Defense
          </option>
          <option
            value="Airlines"
          >
            Airlines
          </option>
          <option
            value="Auto Components"
          >
            Auto Components
          </option>
          <option
            value="Automobiles"
          >
            Automobiles
          </option>
          <option
            value="Banking"
          >
            Banking
          </option>
          <option
            value="Basic Materials"
          >
            Basic Materials
          </option>
          <option
            value="Beverages"
          >
            Beverages
          </option>
          <option
            value="Biotechnology"
          >
            Biotechnology
          </option>
          <option
            value="Building"
          >
            Building
          </option>
          <option
            value="Chemicals"
          >
            Chemicals
          </option>
          <option
            value="Commercial Services & Supplies"
          >
            Commercial Services & Supplies
          </option>
          <option
            value="Communication Services"
          >
            Communication Services
          </option>
          <option
            value="Communications"
          >
            Communications
          </option>
          <option
            value="Construction"
          >
            Construction
          </option>
          <option
            value="Consumer Cyclical"
          >
            Consumer Cyclical
          </option>
          <option
            value="Consumer Defensive"
          >
            Consumer Defensive
          </option>
          <option
            value="Consumer products"
          >
            Consumer products
          </option>
          <option
            value="Distributors"
          >
            Distributors
          </option>
          <option
            value="Diversified Consumer Services"
          >
            Diversified Consumer Services
          </option>
          <option
            value="Electrical Equipment"
          >
            Electrical Equipment
          </option>
          <option
            value="Energy"
          >
            Energy
          </option>
          <option
            value="Financial"
          >
            Financial
          </option>
          <option
            value="Financial Services"
          >
            Financial Services
          </option>
          <option
            value="Food Products"
          >
            Food Products
          </option>
          <option
            value="Health Care"
          >
            Health Care
          </option>
          <option
            value="Healthcare"
          >
            Healthcare
          </option>
          <option
            value="Hotels, Restaurants & Leisure"
          >
            Hotels, Restaurants & Leisure
          </option>
          <option
            value="Industrial Conglomerates"
          >
            Industrial Conglomerates
          </option>
          <option
            value="Industrial Goods"
          >
            Industrial Goods
          </option>
          <option
            value="Industrials"
          >
            Industrials
          </option>
          <option
            value="Insurance"
          >
            Insurance
          </option>
          <option
            value="Leisure Products"
          >
            Leisure Products
          </option>
          <option
            value="Life Sciences Tools & Services"
          >
            Life Sciences Tools & Services
          </option>
          <option
            value="Logistics & Transportation"
          >
            Logistics & Transportation
          </option>
          <option
            value="Machinery"
          >
            Machinery
          </option>
          <option
            value="Marine"
          >
            Marine
          </option>
          <option
            value="Media"
          >
            Media
          </option>
          <option
            value="Metals & Mining"
          >
            Metals & Mining
          </option>
          <option
            value="Packaging"
          >
            Packaging
          </option>
          <option
            value="Paper & Forest"
          >
            Paper & Forest
          </option>
          <option
            value="Pharmaceuticals"
          >
            Pharmaceuticals
          </option>
          <option
            value="Professional Services"
          >
            Professional Services
          </option>
          <option
            value="Real Estate"
          >
            Real Estate
          </option>
          <option
            value="Retail"
          >
            Retail
          </option>
          <option
            value="Road & Rail"
          >
            Road & Rail
          </option>
          <option
            value="Semiconductors"
          >
            Semiconductors
          </option>
          <option
            value="Technology"
          >
            Technology
          </option>
          <option
            value="Telecommunication"
          >
            Telecommunication
          </option>
          <option
            value="Textiles, Apparel & Luxury Goods"
          >
            Textiles, Apparel & Luxury Goods
          </option>
          <option
            value="Tobacco"
          >
            Tobacco
          </option>
          <option
            value="Trading Companies & Distributors"
          >
            Trading Companies & Distributors
          </option>
          <option
            value="Transportation Infrastructure"
          >
            Transportation Infrastructure
          </option>
          <option
            value="Utilities"
          >
            Utilities
          </option>
        </select>
      </div>
      <div
        style={
          Object {
            "marginRight": 20,
          }
        }
      >
        <span
          style={
            Object {
              "color": "var(--light)",
              "fontFamily": "Alata, sans-serif",
              "fontSize": 20,
              "marginRight": 10,
            }
          }
        >
          Country 
        </span>
        <select
          defaultValue="null"
          name="country"
          onChange={[Function]}
          style={
            Object {
              "borderRadius": 8,
            }
          }
        >
          <option
            value="null"
          >
            Select
          </option>
          <option
            value="US"
          >
            US
          </option>
          <option
            value="MX"
          >
            MX
          </option>
          <option
            value="BR"
          >
            BR
          </option>
          <option
            value="GB"
          >
            GB
          </option>
          <option
            value="CN"
          >
            CN
          </option>
          <option
            value="FI"
          >
            FI
          </option>
          <option
            value="NL"
          >
            NL
          </option>
          <option
            value="SG"
          >
            SG
          </option>
          <option
            value="IN"
          >
            IN
          </option>
          <option
            value="CA"
          >
            CA
          </option>
          <option
            value="ZA"
          >
            ZA
          </option>
          <option
            value="BM"
          >
            BM
          </option>
          <option
            value="DE"
          >
            DE
          </option>
          <option
            value="IE"
          >
            IE
          </option>
          <option
            value="CH"
          >
            CH
          </option>
          <option
            value="LU"
          >
            LU
          </option>
          <option
            value="IL"
          >
            IL
          </option>
          <option
            value="HK"
          >
            HK
          </option>
          <option
            value="TW"
          >
            TW
          </option>
          <option
            value="ES"
          >
            ES
          </option>
          <option
            value="FR"
          >
            FR
          </option>
          <option
            value="MC"
          >
            MC
          </option>
          <option
            value="GR"
          >
            GR
          </option>
          <option
            value="KR"
          >
            KR
          </option>
          <option
            value="PR"
          >
            PR
          </option>
          <option
            value="SE"
          >
            SE
          </option>
          <option
            value="AU"
          >
            AU
          </option>
          <option
            value="RU"
          >
            RU
          </option>
          <option
            value="KY"
          >
            KY
          </option>
          <option
            value="CL"
          >
            CL
          </option>
          <option
            value="CO"
          >
            CO
          </option>
          <option
            value="DK"
          >
            DK
          </option>
          <option
            value="BE"
          >
            BE
          </option>
          <option
            value="JP"
          >
            JP
          </option>
          <option
            value="AR"
          >
            AR
          </option>
          <option
            value="UY"
          >
            UY
          </option>
          <option
            value="CZ"
          >
            CZ
          </option>
          <option
            value="VI"
          >
            VI
          </option>
          <option
            value="JE"
          >
            JE
          </option>
          <option
            value="PA"
          >
            PA
          </option>
          <option
            value="TH"
          >
            TH
          </option>
          <option
            value="ID"
          >
            ID
          </option>
          <option
            value="IT"
          >
            IT
          </option>
          <option
            value="TR"
          >
            TR
          </option>
          <option
            value="PH"
          >
            PH
          </option>
          <option
            value="CY"
          >
            CY
          </option>
          <option
            value="GG"
          >
            GG
          </option>
          <option
            value="MO"
          >
            MO
          </option>
          <option
            value="NZ"
          >
            NZ
          </option>
          <option
            value="NO"
          >
            NO
          </option>
          <option
            value="PE"
          >
            PE
          </option>
          <option
            value="CR"
          >
            CR
          </option>
          <option
            value="BS"
          >
            BS
          </option>
          <option
            value="PT"
          >
            PT
          </option>
          <option
            value="SN"
          >
            SN
          </option>
          <option
            value="CI"
          >
            CI
          </option>
          <option
            value="IS"
          >
            IS
          </option>
          <option
            value="MA"
          >
            MA
          </option>
          <option
            value="CW"
          >
            CW
          </option>
          <option
            value="IM"
          >
            IM
          </option>
          <option
            value="GA"
          >
            GA
          </option>
          <option
            value="ZM"
          >
            ZM
          </option>
          <option
            value="CK"
          >
            CK
          </option>
          <option
            value="VG"
          >
            VG
          </option>
          <option
            value="AE"
          >
            AE
          </option>
          <option
            value="JO"
          >
            JO
          </option>
          <option
            value="MT"
          >
            MT
          </option>
          <option
            value="AT"
          >
            AT
          </option>
          <option
            value="PL"
          >
            PL
          </option>
          <option
            value="FO"
          >
            FO
          </option>
          <option
            value="LI"
          >
            LI
          </option>
          <option
            value="GI"
          >
            GI
          </option>
          <option
            value="AZ"
          >
            AZ
          </option>
          <option
            value="FK"
          >
            FK
          </option>
          <option
            value="UA"
          >
            UA
          </option>
          <option
            value="MY"
          >
            MY
          </option>
          <option
            value="MU"
          >
            MU
          </option>
          <option
            value="TC"
          >
            TC
          </option>
          <option
            value="NG"
          >
            NG
          </option>
          <option
            value="MN"
          >
            MN
          </option>
          <option
            value="TG"
          >
            TG
          </option>
          <option
            value="GE"
          >
            GE
          </option>
          <option
            value="VN"
          >
            VN
          </option>
          <option
            value="HU"
          >
            HU
          </option>
          <option
            value="KH"
          >
            KH
          </option>
          <option
            value="SB"
          >
            SB
          </option>
          <option
            value="PG"
          >
            PG
          </option>
          <option
            value="BB"
          >
            BB
          </option>
          <option
            value="BD"
          >
            BD
          </option>
          <option
            value="DO"
          >
            DO
          </option>
          <option
            value="SK"
          >
            SK
          </option>
          <option
            value="AI"
          >
            AI
          </option>
          <option
            value="LT"
          >
            LT
          </option>
          <option
            value="BG"
          >
            BG
          </option>
        </select>
      </div>
      <div
        style={
          Object {
            "marginRight": 20,
          }
        }
      >
        <span
          style={
            Object {
              "color": "var(--light)",
              "fontFamily": "Alata, sans-serif",
              "fontSize": 20,
              "marginRight": 10,
            }
          }
        >
          Exchange 
        </span>
        <select
          defaultValue="null"
          name="exchange"
          onChange={[Function]}
          style={
            Object {
              "borderRadius": 8,
            }
          }
        >
          <option
            value="null"
          >
            Select
          </option>
          <option
            value="NYSE"
          >
            NYSE
          </option>
          <option
            value="NASDAQ"
          >
            NASDAQ
          </option>
          <option
            value="AMEX"
          >
            AMEX
          </option>
          <option
            value="EURONEX"
          >
            EURONEX
          </option>
          <option
            value="TSX"
          >
            TSX
          </option>
          <option
            value="ETF"
          >
            ETF
          </option>
          <option
            value="MUTUAL FUND"
          >
            MUTUAL FUND
          </option>
        </select>
      </div>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "background": "var(--yellow)",
            "borderRadius": 15,
            "borderStyle": "none",
            "color": "var(--white)",
            "fontFamily": "Alata, sans-serif",
            "fontSize": 20,
            "fontWeight": "bold",
            "paddingBottom": 6,
            "paddingTop": 3,
          }
        }
        type="button"
      >
        Apply
      </button>
    </div>
  </div>
  <div
    className="d-flex flex-row justify-content-start flex-wrap"
    style={
      Object {
        "background": "#17A2B8",
        "borderBottomLeftRadius": "20px",
        "borderBottomRightRadius": "20px",
        "marginTop": "0",
        "minHeight": "90px",
        "paddingBottom": "15px",
        "paddingLeft": "45px",
        "paddingRight": "0",
        "paddingTop": "0",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <svg
        height={64}
        viewBox="0 0 58 58"
        width="30%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
        >
          <g
            stroke="#fff"
            strokeWidth="1.5"
            transform="translate(2 1)"
          >
            <circle
              cx="42.601"
              cy="11.462"
              fill="#fff"
              fillOpacity="1"
              r="5"
            >
              <animate
                attributeName="fill-opacity"
                begin="0s"
                calcMode="linear"
                dur="1.3s"
                repeatCount="indefinite"
                values="1;0;0;0;0;0;0;0"
              />
            </circle>
            <circle
              cx="49.063"
              cy="27.063"
              fill="#fff"
              fillOpacity="0"
              r="5"
            >
              <animate
                attributeName="fill-opacity"
                begin="0s"
                calcMode="linear"
                dur="1.3s"
                repeatCount="indefinite"
                values="0;1;0;0;0;0;0;0"
              />
            </circle>
            <circle
              cx="42.601"
              cy="42.663"
              fill="#fff"
              fillOpacity="0"
              r="5"
            >
              <animate
                attributeName="fill-opacity"
                begin="0s"
                calcMode="linear"
                dur="1.3s"
                repeatCount="indefinite"
                values="0;0;1;0;0;0;0;0"
              />
            </circle>
            <circle
              cx="27"
              cy="49.125"
              fill="#fff"
              fillOpacity="0"
              r="5"
            >
              <animate
                attributeName="fill-opacity"
                begin="0s"
                calcMode="linear"
                dur="1.3s"
                repeatCount="indefinite"
                values="0;0;0;1;0;0;0;0"
              />
            </circle>
            <circle
              cx="11.399"
              cy="42.663"
              fill="#fff"
              fillOpacity="0"
              r="5"
            >
              <animate
                attributeName="fill-opacity"
                begin="0s"
                calcMode="linear"
                dur="1.3s"
                repeatCount="indefinite"
                values="0;0;0;0;1;0;0;0"
              />
            </circle>
            <circle
              cx="4.938"
              cy="27.063"
              fill="#fff"
              fillOpacity="0"
              r="5"
            >
              <animate
                attributeName="fill-opacity"
                begin="0s"
                calcMode="linear"
                dur="1.3s"
                repeatCount="indefinite"
                values="0;0;0;0;0;1;0;0"
              />
            </circle>
            <circle
              cx="11.399"
              cy="11.462"
              fill="#fff"
              fillOpacity="0"
              r="5"
            >
              <animate
                attributeName="fill-opacity"
                begin="0s"
                calcMode="linear"
                dur="1.3s"
                repeatCount="indefinite"
                values="0;0;0;0;0;0;1;0"
              />
            </circle>
            <circle
              cx="27"
              cy="5"
              fill="#fff"
              fillOpacity="0"
              r="5"
            >
              <animate
                attributeName="fill-opacity"
                begin="0s"
                calcMode="linear"
                dur="1.3s"
                repeatCount="indefinite"
                values="0;0;0;0;0;0;0;1"
              />
            </circle>
          </g>
        </g>
      </svg>
    </div>
  </div>
</div>
`;
